src/components/CourseList/CourseItem.js

```diff
@@ -2,15 +2,27 @@ import React, {PropTypes} from 'react';
 import {connect} from 'react-redux';
 import styles from './CourseItem.scss';
 import withStyles from 'isomorphic-style-loader/lib/withStyles';
-
+import {getLessonIntro} from '../../util';
 import Link from 'react-router/lib/Link';
 import Glyphicon from 'react-bootstrap/lib/Glyphicon';
 import {getTranslator} from '../../selectors/translate';
+import Tooltip from 'react-bootstrap/lib/Tooltip';
+import OverlayTrigger from 'react-bootstrap/lib/OverlayTrigger';
 
-const CourseItem = ({course, t}) => {
+const CourseItem = ({course, t, language}) => {
+  const coursePath = course.name.replace(/ /g, '_').toLowerCase();
   const isExternal = course.hasOwnProperty('externalLink');
+  const tooltipContent = isExternal ? getLessonIntro(coursePath + '/index') :
+    getLessonIntro(coursePath + '/index' + (language === 'nb' ? '' : ('_' + language)));
+  const createMarkup = () => {
```

Comment:
Could we put all this logic into a function and/or selector which just takes the course name, and returns the markup object? It seems there is a similar need for lesson?